### PR TITLE
Lint Community Scripts

### PR DESCRIPTION
**Pull request by Trenly**
Lints the community scripts and takes care of the ScriptAnalyzer issues

@denelon
@mdanish-kh
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/132425)

### CODE REVIEW

File: Tools/YamlCreate.ps1
Comment (mdanish-kh) [inline Tools/YamlCreate.ps1:1]: Seeing some warnings on YamlCreate (if you wish to handle those) -

```
 ~/winget-pkgs pr/Trenly/132425 Invoke-ScriptAnalyzer .\Tools\YamlCreate.ps1

RuleName                            Severity     ScriptName Line  Message
--------                            --------     ---------- ----  -------
PSUseBOMForUnicodeEncodedFile       Warning      YamlCreate       Missing BOM encoding for non-ASCII encoded file
                                                 .ps1             'YamlCreate.ps1'
PSUseSingularNouns                  Warning      YamlCreate 546   The cmdlet 'Get-ItemMetadata' uses a plural noun. A
                                                 .ps1             singular noun should be used instead.
PSUseSingularNouns                  Warning      YamlCreate 1586  The cmdlet 'Read-InstallerMetadata' uses a plural noun. A
                                                 .ps1             singular noun should be used instead.
PSUseSingularNouns                  Warning      YamlCreate 1666  The cmdlet 'Read-LocaleMetadata' uses a plural noun. A
                                                 .ps1             singular noun should be used instead.
```

**Review by mdanish-kh — APPROVED** [review]
Out of curiosity - did you manage to get these warnings shown in VSCode? I could see others if I make a manual error in the code, but couldn't manage to see those that you handled in the code. Running `Invoke-ScriptAnalyzer` works tho:

image

**Comment by denelon** [comment]
@Trenly did you want to update this one, or merge it and work on @mdanish-kh suggestion/question?

**Comment by Trenly** [comment]
> @Trenly did you want to update this one, or merge it and work on @mdanish-kh suggestion/question?

Wait a min and I'll see what I can find

File: Tools/YamlCreate.ps1
Comment (Trenly) [reply Tools/YamlCreate.ps1:1]: Interesting . . . only the PSUseBOMForUnicodeEncodedFile shows up when I run it

Comment (Trenly) [reply Tools/YamlCreate.ps1:1]: @mdanish-kh - Seems it's a difference between PowerShell 7 and PowerShell; Will address

**Review by mdanish-kh — APPROVED** [review]
All warnings resolved

**Comment by Trenly** [comment]
cc @denelon

**Review by denelon — APPROVED** [review]

**Pull request merged**